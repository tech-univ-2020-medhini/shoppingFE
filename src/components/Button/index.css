.button-yellow{
    background-color: rgb(238,195,92);
    height: 2rem;
    width: 10em;
    color: white;
    font-size: 1.1rem;
    border-radius: 6px;
    margin: 1rem;
}
.button-white{
    background-color: rgb(255, 255, 255);
    height: 2rem;
    width: 10em;
    color: white;
    font-size: 1.1rem;
    border-radius: 6px;
    margin: 1rem;
    color: black;
}