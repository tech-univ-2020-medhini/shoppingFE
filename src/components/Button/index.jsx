import React, { useState } from 'react';
import PropTypes from 'prop-types';
import * as styles from './index.css';

const Button = (props) => {
  const { text, buttonColor, onClickHandler } = props;
  return (
    <button className={buttonColor === 'yellow' ? styles.yellow : styles.grey} onClick={onClickHandler} type="button">{text}</button>
  );
};
Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClickHandler: PropTypes.func.isRequired,
  buttonColor: PropTypes.string.isRequired,
};
export default Button;
