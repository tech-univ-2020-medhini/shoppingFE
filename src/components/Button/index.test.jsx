import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';


describe('the button', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button />);

    expect(asFragment()).toMatchSnapshot();
  });
});
