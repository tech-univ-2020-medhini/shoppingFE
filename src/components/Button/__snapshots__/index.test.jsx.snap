// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button should render correctly 1`] = `
<DocumentFragment>
  <button
    type="button"
  />
</DocumentFragment>
`;
